### ISSUE TITLE
RTL doesn't update the link register on JAL instruction

### ISSUE THREAD

**Issue by guptasanchi** [SimpleCPU/SimpleCPU#21]
After the fix for #20 , the test seems to fail with the following mismatch - 
```
# PC:00000008	INSTR:0c000222	 JAL 888     
# [RTL]  	PC:00000008	Instr:0c000222	R0:00000000	R0:00000000
# [MODEL]	PC:00000008	Instr:0c000222	R0:00000000	R31:0000000c
# Unexpected R0 Register
# Expecting  R31 Register
# ** Fatal: TEST FAILED
```

The model updates the link reg (R31) on JAL instruction but RTL doesn't seem to do that.

**Comment by raulbehl** [comment]
There was some issue with the current decoding of the "use_link_reg_dec" signal. I have updated the logic to decode J-type instructions correctly. Fix in a595612

**Comment by raulbehl** [comment]
Fix in master.
